src/float/printer/hexfloat.cr

```diff
@@ -0,0 +1,256 @@
+module Float::Printer::Hexfloat(F, U)
+  # IEEE defines the following grammar:
+  #
+  # ```
+  # sign            = /[+−]/
+  # digit           = /[0123456789]/
+  # hexDigit        = /[0123456789abcdefABCDEF]/
+  # hexExpIndicator = /[Pp]/
+  # hexIndicator    = /0[Xx]/
+  # hexSignificand  = /#{hexDigit}*\.#{hexDigit}+|#{hexDigit}+\.|#{hexDigit}+/
+  # decExponent     = /#{hexExpIndicator}#{sign}?#{digit}+/
+  # hexSequence     = /#{sign}?#{hexIndicator}#{hexSignificand}#{decExponent}/
+  #                 = /[+−]?0[Xx](?:[0-9A-Fa-f]*\.[0-9A-Fa-f]+|[0-9A-Fa-f]+\.?)[Pp][+−]?[0-9]+/
+  # ```
+  @[AlwaysInline]
+  def self.to_f(str : String, &)
```

Comment:
**Thought:** This method yields so it will always be inlined regardless of the annotation. Does it still make sense to have it for explicitness? 🤔

Reply:
Apparently this happens to both definitions of `Crystal::System::FileDescriptor#system_console_mode(&)` as well